### PR TITLE
ErrorVector::plot_error Nemesis support

### PR DESCRIPTION
**Pull request by roystgnr**
Plotting error indicator output is often nice.  Plotting it to a
standard format that viz tools can read is nice.  Being unable to do
that on a distributed mesh without serializing first is not nice, so
let's fix that.

### CODE REVIEW

**Comment by roystgnr** [comment]
Huh.  Wonder why my runs didn't trigger *that*...

**Comment by roystgnr** [comment]
This actually seems to be exposing a preexisting bug.  Not a very serious one, just another "more processors than elements" corner case.  I'll throw in the fix here when I'm done with it.

File: src/mesh/nemesis_io_helper.C [hunk 1]
```diff
@@ -621,11 +621,14 @@ void Nemesis_IO_Helper::put_node_cmap(std::vector<std::vector<int>> & node_cmap_
...
+        nullptr : &node_cmap_node_ids_in[i][0];
+      int * proc_ids_ptr = node_cmap_proc_ids_in[i].empty() ?
+        nullptr : &node_cmap_proc_ids_in[i][0];
+
       nemesis_err_flag =
-        Nemesis::ne_put_node_cmap(ex_id,
-                                  this->node_cmap_ids[i],
-                                  &node_cmap_node_ids_in[i][0],
-                                  &node_cmap_proc_ids_in[i][0],
+        Nemesis::ne_put_node_cmap(ex_id, this->node_cmap_ids[i],
```
Comment (jwpeterson) [inline src/mesh/nemesis_io_helper.C]: Is it definitely safe to call `ne_put_node_cmap` with `nullptr` inputs? It looks like we weren't even guarding against accessing `[0]` of an empty vector before, so this is definitely an improvement.

File: src/mesh/nemesis_io_helper.C [hunk 2]
```diff
@@ -574,18 +574,18 @@ void Nemesis_IO_Helper::put_cmap_params(std::vector<int> & node_cmap_ids_in,
...
-                                    &node_cmap_ids_in[0],
-                                    &node_cmap_node_cnts_in[0],
-                                    &elem_cmap_ids_in[0],
-                                    &elem_cmap_elem_cnts_in[0],
-                                    this->processor_id());
-    }
+  libmesh_assert(!node_cmap_ids_in.empty());
+  libmesh_assert(!node_cmap_node_cnts_in.empty());
+  libmesh_assert(!elem_cmap_ids_in.empty());
+  libmesh_assert(!elem_cmap_elem_cnts_in.empty());
```
Comment (jwpeterson) [inline src/mesh/nemesis_io_helper.C]: Maybe these should be `libmesh_error`s instead?

File: src/mesh/nemesis_io_helper.C (hunk 1 above)
Comment (roystgnr) [reply src/mesh/nemesis_io_helper.C]: Definitely safe, as in there's a well-documented standard which says we can do this?  No.

Probably safe, as in I've hit a corner case which passes nullptr and our contrib Exodus was okay with that?  Yes.

File: src/mesh/nemesis_io_helper.C (hunk 2 above)
Comment (roystgnr) [reply src/mesh/nemesis_io_helper.C]: Nah, this is an internal consistency check.  The library itself will (now) never trigger it, and it's in the write path not read path, so there's no way for a user to trigger this with a corrupt file or anything other than by manually misusing a low-level method.

**Comment by roystgnr** [comment]
DistributedMesh is also happy with it on up to 24 processors.

**Pull request merged**